File: src/classes/api/req_res/update-lantana-request.ts
'use strict'

import { Lantana } from '@/classes/datas/lantana'
import { GkillAPIRequest } from '../gkill-api-request'
import { Kyou } from '@/classes/datas/kyou'

export class UpdateLantanaRequest extends GkillAPIRequest {

    lantana: Lantana

    tx_id: string | null = null

    want_response_kyou: boolean

    updated_kyou: Kyou | null | null = null

    constructor() {
        super()
        this.lantana = new Lantana()
        this.want_response_kyou = false
    }

}

File: src/app/gkill/api/req_res/update_lantana_request.go
package req_res

import "github.com/mt3hr/gkill/src/app/gkill/dao/reps"

type UpdateLantanaRequest struct {
	SessionID string `json:"session_id"`

	Lantana *reps.Lantana `json:"lantana"`

	TXID *string `json:"tx_id"`

	LocaleName string `json:"locale_name"`

	UpdatedKyou *reps.Kyou `json:"updated_kyou"`

	WantResponseKyou bool `json:"want_response_kyou"`
}
